### PR TITLE
First draft

### PR DESCRIPTION
**Pull request by spencergilbert**
Rough first draft of what a "chart per component" design could look like, based on a combination of the existing Thanos charts we identified here: https://docs.google.com/document/d/18GXxwOm9c2fDK20LV1Sekte0xvK4qXpIGwBcuw_ZYSA/edit#

### CODE REVIEW

File: charts/thanos-store/templates/ingress.yaml [hunk 1]
```diff
@@ -0,0 +1,87 @@
+{{- if .Values.http.ingress.enabled -}}
+{{- $fullName := include "thanos-query.fullname" . -}}
```
Comment (spencergilbert) [inline charts/thanos-store/templates/ingress.yaml:2]: Was pointed out in slack that I didn't copy-paste this well :smile:

**Comment by spencergilbert** [comment]
My biggest/only concern is if we should do a separate chart per Thanos component, or if they should all be parts of a single Thanos chart

File: charts/thanos-store/templates/ingress.yaml (hunk 1 above)
Comment (spencergilbert) [reply charts/thanos-store/templates/ingress.yaml]: Fixed this and a few other cases in the template.

**Comment by ChiefAlexander** [comment]
>My biggest/only concern is if we should do a separate chart per Thanos component

Stumbled upon this while I was looking at implementing Thanos, throwing out my own thoughts. Ideally, the end state that I would expect is that when I run a helm install all of the components that I need to run Thanos are present. Maybe the path here is to utilize helm dependences in a "main" chart that can fulfill that role, while still allowing users to utilize the underlying component charts.

Also if the path that we are going to take here is to use multiple charts I would suggest utilizing a helm library chart -> https://helm.sh/docs/topics/library_charts/

**Comment by spencergilbert** [comment]
I'm going to close this in favor of PRs per component.

**Pull request closed without merging**